waku/v2/protocol/waku_filter_v2/protocol.nim

```diff
@@ -125,8 +119,61 @@ proc handleSubscribeRequest*(wf: WakuFilter, peerId: PeerId, request: FilterSubs
   else:
     return FilterSubscribeResponse.ok(request.requestId)
 
-proc handleMessage*(wf: WakuFilter, message: WakuMessage) =
-  raiseAssert "Unimplemented"
+proc pushToPeer(wf: WakuFilter, peer: PeerId, buffer: seq[byte]) {.async.} =
+  trace "pushing message to subscribed peer", peer=peer
+
+  if not wf.peerManager.peerStore.hasPeer(peer, WakuFilterPushCodec):
+    # Check that peer has not been removed from peer store
```

Comment:
Beyond this pr, but this makes me think that the peermanager should have knowledge of subscriptions/filter to not remove these peers from the peerstore.

Reply:
I agree! My current thinking is that we may implement some kind of policy which could be set by clients of the peer store. Needs more thinking, though.

Reply:
> the peermanager should have knowledge of subscriptions/filter to not remove these peers from the peerstore.

It makes sense, but... How would you implement that without coupling the peer management code to the filter protocol code?

Reply:
> It makes sense, but... How would you implement that without coupling the peer management code to the filter protocol code?

mm interesting. I would say its not possible without coupling it. I mean, the cleanest thing imho would be to have a new field in the Peer (subscribedToFilter) that can be set or unset via a proc used by filter. Then the peer manager will never? drop this connection.